core/src/no/group15/playmagic/ecs/systems/MovementSystem.kt

```diff
@@ -116,4 +120,14 @@ class MovementSystem(
 		}
 	}
 
+
+	override fun receive(signal: Signal<CollisionEvent>, event: CollisionEvent) {
+		val destructible = mapperFor<DestructibleComponent>()
+		val rock = if (event.entity1.has(destructible)) event.entity1 else event.entity2
+		val rockPosition = rock[transformMapper]!!.position
```

Comment:
these should be the opposite `transformMapper[rock].position`
always `mapper[entity]`

Reply:
Do you mean as best practice? Because also that syntax  work thanks to KTX

Reply:
yes, and you don't have to use all the exclamation marks
I've changed these in master

Reply:
It's like you're looking up something in a map, I just think it makes more sense that way 😄

Reply:
make sense. I seen that syntax in KTX and I just started using it, I liked it because if you call the mapper `transform` you can access to the component with this syntax `entity[transform]` (it is cool). But the fact you have to use the double exclamation mark does it make all horrible. I agree with you at the end :+1:

Reply:
but calling the mapper transform is very confusing, it should be named for what it is; a transformMapper. transform is what the mapper returns.
player is even worse 😄 
`val player = player[player]!!`
like what is what